src/Request.ts

```diff
@@ -0,0 +1,350 @@
+	/**
+	 * A flag indicating whether the target contains a query string
+	 */
+	protected _targetContainsQueryString: boolean;
+
+	/**
+	 * Defines the Accept header to use on HTTP requests
+	 */
+	accepts: string;
+
+	/**
+	 * The prefix to apply to sort property names that are ascending
+	 */
+	ascendingPrefix: string;
+
+	/**
+	 * The prefix to apply to sort property names that are ascending
+	 */
+	descendingPrefix: string;
+
+	/**
+	 * Additional headers to pass in all requests to the server. These can be overridden
+	 * by passing additional headers to calls to the store.
+	 */
+	headers: {};
+
+	/**
+	 * The indicates if range limits (start and end) should be specified
+	 * in a query parameter, and what the start parameter should be.
+	 * This must be used in conjunction with the rangeCountParam
+	 * If this is not specified, the range will
+	 * included with a RQL style limit() parameter
+	 */
+	rangeStartParam: string;
+
+	/**
+	 * The indicates if range limits (start and end) should be specified
+	 * in a query parameter, and what the count parameter should be.
+	 * This must be used in conjunction with the rangeStartParam
+	 * If this is not specified, the range will
+	 * included with a RQL style limit() parameter
+	 */
+	rangeCountParam: string;
+
+	/**
+	 * This will specify the query parameter to use for specifying the
+	 * 'select` properties. This will default to `select(<properties>)`
+	 * in the query string.
+	 */
+	selectParam: string;
+
+	/**
+	 * The query parameter to used for holding sort information. If this is omitted, than
+	 * the sort information is included in a functional query token to avoid colliding
+	 * with the set of name/value pairs.
+	 */
+	sortParam: string;
+
+	/**
+	 * The target base URL to use for all requests to the server. This string will be
+	 * prepended to the id to generate the URL (relative or absolute) for requests
+	 * sent to the server
+	 */
+	target: string;
+
+	/**
+	 * The indicates if range limits (start and end) should be specified
+	 * a Range header, using items units. If this is set to true, a header
+	 * be included of the form:
+	 * Range: items=start-end
+	 */
+	useRangeHeaders: boolean;
+
+	/**
+	 * This is a basic store for RESTful communicating with a server through JSON
+	 * formatted data. It extends dstore/Store.
+	 * @param options This provides any configuration information that will be mixed into the store
+	 */
+	constructor(options?: RequestStoreArgs) {
+		super(options);
+		/**
+		 * Defaults to JSON, but other formats can be parsed by providing an alternate
+		 * parsing function. If you do want to use an alternate format, you will probably
+		 * want to use an alternate stringify function for the serialization of data as well.
+		 * Also, if you want to support parsing a larger set of JavaScript objects
+		 * outside of strict JSON parsing, you can provide dojo/_base/json.fromJson as the parse function
+		 */
+		this.parse = this.parse || function (data: string) {
```

Comment:
If we make `parse` a method on `Store`, we ought to be able to make the default a method here too, and just naturally allow `options` to naturally override it via `mixin`.